### PR TITLE
Added an option to add a delay to threads

### PR DESCRIPTION
**Pull request by 0x6368**

### CODE REVIEW

File: cli/cmd/root.go [hunk 1]
```diff
@@ -109,6 +119,7 @@ func configureGlobalOptions() {
 }
 
 func init() {
+	rootCmd.PersistentFlags().IntP("delay", "d", 0, "Number of milliseconds each thread waits between requests")
```
Comment (FireFart) [inline cli/cmd/root.go]: this can be a `DurationP` so you don't need to convert it

File: libgobuster/libgobuster.go [hunk 2]
```diff
@@ -113,6 +114,8 @@ func (g *Gobuster) worker(wordChan <-chan string, wg *sync.WaitGroup) {
 					g.resultChan <- r
 				}
 			}
+
+			time.Sleep(time.Duration(g.Opts.Delay) * time.Millisecond)
```
Comment (FireFart) [inline libgobuster/libgobuster.go]: this should be smth like this so ctrl+c works correctly:
```golang
select {
  case <-g.context.Done():
    return
  case <-time.After(g.Opts.Delay):
    return
}
```

File: libgobuster/options.go [hunk 3]
```diff
@@ -10,6 +10,7 @@ type Options struct {
 	Quiet          bool
 	WildcardForced bool
 	Verbose        bool
+	Delay          int
```
Comment (FireFart) [inline libgobuster/options.go]: This should be a `time.Duration`

File: gobusterdir/gobusterdir.go [hunk 4]
```diff
@@ -242,10 +242,15 @@ func (d *GobusterDir) GetConfigString() (string, error) {
 	if _, err := fmt.Fprintf(tw, "[+] Url:\t%s\n", o.URL); err != nil {
 		return "", err
 	}
+
 	if _, err := fmt.Fprintf(tw, "[+] Threads:\t%d\n", d.globalopts.Threads); err != nil {
 		return "", err
 	}
 
+	if _, err := fmt.Fprintf(tw, "[+] Delay:\t%dms\n", d.globalopts.Delay); err != nil {
```
Comment (FireFart) [inline gobusterdir/gobusterdir.go]: we should only print this out if it is set

File: gobusterdns/gobusterdns.go [hunk 5]
```diff
@@ -173,6 +173,10 @@ func (d *GobusterDNS) GetConfigString() (string, error) {
 		return "", err
 	}
 
+	if _, err := fmt.Fprintf(tw, "[+] Delay:\t%d\n", d.globalopts.Delay); err != nil {
```
Comment (FireFart) [inline gobusterdns/gobusterdns.go]: we should only print this out if it is set

File: gobustervhost/gobustervhost.go [hunk 6]
```diff
@@ -143,10 +143,15 @@ func (v *GobusterVhost) GetConfigString() (string, error) {
 	if _, err := fmt.Fprintf(tw, "[+] Url:\t%s\n", o.URL); err != nil {
 		return "", err
 	}
+
 	if _, err := fmt.Fprintf(tw, "[+] Threads:\t%d\n", v.globalopts.Threads); err != nil {
 		return "", err
 	}
 
+	if _, err := fmt.Fprintf(tw, "[+] Delay:\t%d\n", v.globalopts.Delay); err != nil {
```
Comment (FireFart) [inline gobustervhost/gobustervhost.go]: we should only print this out if it is set

**Comment by FireFart** [comment]
please also create a feature branch for your pull request instead of creating it from `master`

**Comment by 0x6368** [comment]
Thanks for the feedback!
I changed the delay type to time.Duration and created a new pull request from a feature branch: https://github.com/OJ/gobuster/pull/153

**Pull request closed without merging**